File: src/test/java/cloud/cosmin/checklister/service/ConverterServiceTest.java
package cloud.cosmin.checklister.service;

import cloud.cosmin.checklister.dao.ItemEntity;
import cloud.cosmin.checklister.dao.ListEntity;
import cloud.cosmin.checklister.dto.ItemGetDto;
import cloud.cosmin.checklister.dto.ListGetDto;
import org.junit.Test;

import java.util.UUID;

import static org.junit.Assert.*;

public class ConverterServiceTest {
    @Test
    public void testConvert() {
        ConverterService svc = new ConverterService();

        ListEntity entity = new ListEntity();
        entity.setId(UUID.randomUUID());
        entity.setTitle("title");

        ListGetDto dto = svc.listDto(entity);
        assertEquals(entity.getId(), dto.getId());
        assertEquals("title", dto.getTitle());
    }

    @Test
    public void itemConvert() {
        ConverterService svc = new ConverterService();

        ListEntity list = new ListEntity();
        list.setId(UUID.randomUUID());
        list.setTitle("title");

        ItemEntity item = new ItemEntity();
        item.setId(UUID.randomUUID());
        item.setList(list);
        item.setContent("content");
        item.setContentType("text/plain");
        item.setRank(0);

        ItemGetDto itemDto = svc.itemDto(item);
        assertEquals(item.getId(), itemDto.getId());
        assertEquals(list.getId(), itemDto.getList());
        assertEquals("content", itemDto.getContent());
        assertEquals("text/plain", itemDto.getContentType());
        assertEquals(Integer.valueOf(0), itemDto.getRank());
    }
}

File: src/test/kotlin/cloud/cosmin/checklister/service/ConverterServiceTest.kt
package cloud.cosmin.checklister.service

import cloud.cosmin.checklister.dao.ItemEntity
import cloud.cosmin.checklister.dao.ListEntity
import org.junit.Assert.assertEquals
import org.junit.Test
import java.util.*

class ConverterServiceTest {
    @Test
    fun testConvert() {
        val svc = ConverterService()

        val entity = ListEntity()
        entity.id = UUID.randomUUID()
        entity.title = "title"

        val (id, title) = svc.listDto(entity)
        assertEquals(entity.id, id)
        assertEquals("title", title)
    }

    @Test
    fun itemConvert() {
        val svc = ConverterService()

        val list = ListEntity()
        list.id = UUID.randomUUID()
        list.title = "title"

        val item = ItemEntity()
        item.id = UUID.randomUUID()
        item.list = list
        item.content = "content"
        item.contentType = "text/plain"
        item.rank = 0

        val itemDto = svc.itemDto(item)
        assertEquals(item.id, itemDto.id)
        assertEquals(list.id, itemDto.list)
        assertEquals("content", itemDto.content)
        assertEquals("text/plain", itemDto.contentType)
        assertEquals(Integer.valueOf(0), itemDto.rank)
    }
}